src/actions.ts

```diff
@@ -122,53 +122,65 @@ export function copy<T>(state: Types.StoreState<T>) {
   };
 }
 
-export const cut = (state: Types.StoreState<*>) => ({
+export const cut = (state: Types.IStoreState<any>) => ({
   ...copy(state),
   cut: true
 });
 
-export function paste<Cell: Types.CellBase>(state: Types.StoreState<Cell>) {
+export function paste<Cell>(state: Types.IStoreState<Cell>) {
   if (PointSet.isEmpty(state.copied)) {
     return null;
   }
   const minPoint = PointSet.min(state.copied);
 
-  type Accumulator = {|
-    data: $PropertyType<Types.StoreState<Cell>, "data">,
-    selected: $PropertyType<Types.StoreState<Cell>, "selected">,
-    commit: $PropertyType<Types.StoreState<Cell>, "lastCommit">
-  |};
+  type Accumulator = {
+    data: Types.IStoreState<Cell>["data"];
+    selected: Types.IStoreState<Cell>["selected"];
+    commit: Types.IStoreState<Cell>["lastCommit"];
+  };
 
   const { data, selected, commit } = PointMap.reduce(
-    (acc: Accumulator, value, { row, column }): Accumulator => {
+    (
+      acc: Accumulator,
+      value:
+        | Types.CellChange<Cell>["prevCell"]
+        | Types.CellChange<Cell>["nextCell"],
+      { row, column }
+    ): any => {
       if (!state.active) {
         return acc;
       }
 
-      let commit =
-        acc.commit || ([]: $PropertyType<Types.StoreState<Cell>, "lastCommit">);
+      let commit: null | Types.IStoreState<Cell>["lastCommit"] =
```

Comment:
Unclear why `null` was added here when there's an `|| []` in the assign. `null` would never be the value. This also adds additional logic statement requirements below to check the presence of `commit`